cmake/modules/configuration_files.cmake

```
@@ -77,8 +77,20 @@ zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards DTS APP_BOARD_DTS SUFFIX
 
 zephyr_get(DTC_OVERLAY_FILE SYSBUILD LOCAL)
 if(NOT DEFINED DTC_OVERLAY_FILE)
+  set(board_with_identifier_overlay)
+
+  if(DEFINED BOARD_IDENTIFIER)
+    # Generate overlay filename of board with identifier present
+    zephyr_build_string(board_with_identifier_overlay
+                        BOARD ${BOARD}
+                        BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
+    )
+
+    set(board_with_identifier_overlay "${board_with_identifier_overlay}.overlay")
+  endif()
```

How about:
```suggestion
  zephyr_build_string(board_overlay_strings
                      BOARD ${BOARD}
                      BOARD_IDENTIFIER ${BOARD_IDENTIFIER}
                      MERGE
  )
  list(TRANSFORM board_overlay_strings APPEND ".overlay")
```
and then use `board_overlay_strings` below in:

```
        NAMES "${APP_BOARD_DTS};${board_overlay_strings };app.overlay" SUFFIX ${FILE_SUFFIX})
```

---

Fixed